stdpopsim/generic_models.py

```diff
@@ -0,0 +1,87 @@
+# generic_models
+import msprime
+import stdpopsim.models as models
+
+
+#########
+# Generic Demographic Models
+#
+#
+########
+
+class generic_constant_size(models.Model):
+    """
+    Model Name:
+        generic_constant_size
+
+    Model Description:
+        single-size, single-population model
+
+    Model population indexes:
+        - Population1 : 0
+
+    Parameter Table:
+        .. csv-table::
+            :widths: 15 8 20
+            :header: "Parameter Type (units)", "Value", "Description"
+            :file: ../docs/parameter_tables/homo_sapiens/GutenkunstThreePopOutOfAfrica_params.csv
+
+    CLI help:
+        python -m stdpopsim homo-sapiens generic_constant_size -h
+
+    Citation:
+
+    """  # noqa: E501
+
+    def __init__(self, N=100):
+        """
+        Initialize constant size model with
+        population size N
+        """
+        super().__init__()
+        self.population_configurations = [
+            msprime.PopulationConfiguration(initial_size=N)
+        ]
+        self.migration_matrix = [[0]]
+        self.demographic_events = []
+
+
+class generic_two_epoch(models.Model):
+    """
+    Model Name:
+        generic_two_epoch
+
+    Model Description:
+        two epoch population size, single-population model
+
+    Model population indexes:
+        - Population1 : 0
+
+    Parameter Table:
+        .. csv-table::
+            :widths: 15 8 20
+            :header: "Parameter Type (units)", "Value", "Description"
+            :file: ../docs/parameter_tables/homo_sapiens/GutenkunstThreePopOutOfAfrica_params.csv
+
+    CLI help:
+        python -m stdpopsim homo-sapiens generic_two_epoch -h
+
+    Citation:
+
+    """  # noqa: E501
+
+    def __init__(self, N1=1000, N2=100, t=10):
```

Comment:
Ditto with the defaults here.

Reply:
But yeah, how do we deal with situations like this? Maybe the tests using ``all_models`` are just over generalising, and we should ditch it? Some things will need parameters, and putting in fake ones by default just to please a test case doesn't seem like the right approach to me.

Reply:
having this limited to the generic cases doesn't bother me much and having the tests that use `all_models` are important. can you see a clean solution here?